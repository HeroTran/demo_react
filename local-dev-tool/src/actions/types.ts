/**
 * Type Action For Home Magazine
 */
export const SHOW_LIST_SITE = 'SHOW_LIST_SITE';
export const SHOW_MODAL = 'SHOW_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const ON_CHANGE_INPUT = 'ON_CHANGE_INPUT';
export const SUBMIT_CONFIG = 'SUBMIT_CONFIG';
export const DELETE_SITE = 'DELETE_SITE';
export const SEARCH_SITE = "SEARCH_SITE";

/**
 * Type Action For site magazine
 */

export const SHOW_ALL_SITE = 'SHOW_ALL_SITE';
export const SHOW_FLAT_FORM = 'SHOW_FLAT_FORM';
export const SHOW_RELOAD_PAGE = 'SHOW_RELOAD_PAGE';
export const EVENT_DOWNLOAD = 'EVENT_DOWNLOAD';
export const EVENT_VIEW = 'EVENT_VIEW';
export const EVENT_DELETE = 'EVENT_DELETE';
export const EVENT_CONFIG = 'EVENT_CONFIG';
export const SHOW_TAB = 'SHOW_TAB';
export const OPEN_MODAL = "OPEN_MODAL";
export const CLOSE_MODAL_CONFIG = "CLOSE_MODAL_CONFIG";
export const SEARCH_ISSUE = "SEARCH_ISSUE";
export const CHANGE_LOADDING = "CHANGE_LOADDING";

/**
 * Type Action For site magazine
 */

export const SHOW_ALL_ISSUE = 'SHOW_ALL_ISSUE';
export const SEARCH_STORY = 'SEARCH_STORY';
export const BUTTON_CHANGE_FLATFORM = 'BUTTON_CHANGE_FLATFORM';
export const BUTTON_CHANGE_MODE = 'BUTTON_CHANGE_MODE';
export const BUTTON_CHANGE_WIDTH = 'BUTTON_CHANGE_WIDTH';
export const BUTTON_CHANGE_SIZE = 'BUTTON_CHANGE_SIZE';
export const ZOOM_FRAME = 'ZOOM_FRAME';
export const CHANGE_URL_IFRAME = 'CHANGE_URL_IFRAME';
export const CHANGE_LAYOUT = 'CHANGE_LAYOUT';
export const CHANGE_SWIPE = 'CHANGE_SWIPE';

export const DOWNLOAD_SUCCESS = 'DOWNLOAD_SUCCESS';
export const DOWNLOAD_REQUEST = 'DOWNLOAD_REQUEST';
export const DOWNLOAD_FAILED = 'DOWNLOAD_FAILED';

export const DELETED_ISSUE_SUCCESS = 'DELETED_ISSUE_SUCCESS';
export const DELETED_ISSUE_REQUEST = 'DELETED_ISSUE_REQUEST';
export const DELETED_ISSUE_FAILED = 'DELETED_ISSUE_FAILED';

/**
 * Fetch Data
 * 
 */

export const FETCH_DATA_SUCCCESS_MAGAZINE = "FETCH_DATA_SUCCCESS_MAGAZINE";
export const FETCH_DATA_REQUESTED_MAGAZINE = "FETCH_DATA_REQUESTED_MAGAZINE";
export const FETCH_DATA_FAILED_MAGAZINE = "FETCH_DATA_FAILED_MAGAZINE";

export const FETCH_DATA_SUCCCESS_SITE = "FETCH_DATA_SUCCCESS_SITE";
export const FETCH_DATA_REQUESTED_SITE = "FETCH_DATA_REQUESTED_SITE";
export const FETCH_DATA_FAILED_SITE = "FETCH_DATA_FAILED_SITE";


export const FETCH_DATA_SUCCCESS_ISSUE = "FETCH_DATA_SUCCCESS_ISSUE";
export const FETCH_DATA_REQUESTED_ISSUE = "FETCH_DATA_REQUESTED_ISSUE";
export const FETCH_DATA_FAILED_ISSUE = "FETCH_DATA_FAILED_ISSUE";

export const FETCH_INFO_CONFIG_SUCCESS = "FETCH_INFO_CONFIG_SUCCESS";
export const FETCH_INFO_CONFIG_REQUEST = "FETCH_INFO_CONFIG_REQUESTED";
export const FETCH_INFO_CONFIG_FAILED = "FETCH_INFO_CONFIG_FAILED";