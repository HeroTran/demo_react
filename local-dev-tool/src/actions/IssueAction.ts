import * as types from './types';
import * as api from '../api/cms';
import * as handleMethod from '../utility/handleMethod';


export function showInfoIssue(params) {
    return (dispatch, getState) => {
        var promise = api.getInfoIssues(params.siteId, params.issueId);
        dispatch(getListSiteRequest("waiting fecth for Issue..."));
        return promise.then((result) => {
            dispatch(getListSiteSuccess(result, true));
        })
        .catch(e => {
            dispatch(getListSiteFailed("Failed fecth for Issue...."))
        })
    }
}

export function getListSiteSuccess(issue, isLoadding) {
    return {
        type: types.FETCH_DATA_SUCCCESS_ISSUE,
        payload: {
            issue: issue,
            isLoadding: isLoadding
        }
    };
}

export function getListSiteRequest(isRequest) {
    return {
        type: types.FETCH_DATA_REQUESTED_ISSUE,
        payload: isRequest
    };
}

export function getListSiteFailed(isFailed) {
    return {
        type: types.FETCH_DATA_FAILED_ISSUE,
        payload: isFailed
    };
}

export function getListConfig(params){
    var designPack = handleMethod.getDesignPack(params);
    return (dispatch, getState) => {
        var promise = api.getInfoConfig(designPack);
        dispatch(getInfoConfigRequest("waiting fecth for Info Config..."));
        return promise.then((result) => {
            dispatch(getInfoConfigSuccess(handleMethod.xmlToArray(result),designPack));
        })
        .catch(e => {
            dispatch(getListSiteFailed("Failed fecth for Info Config...."))
        })
    }
}




export function getInfoConfigSuccess(result, designPack) {
    return {
        type: types.FETCH_INFO_CONFIG_SUCCESS,
        payload: {
            result: result,
            designPack: designPack
        }
    };
}

export function getInfoConfigRequest(isRequest) {
    return {
        type: types.FETCH_INFO_CONFIG_REQUEST,
        payload: isRequest
    };
}

export function getInfoConfigFailed(isFailed) {
    return {
        type: types.FETCH_INFO_CONFIG_FAILED,
        payload: isFailed
    };
}

export function changeTemplate(nameLayout){
    return {
        type:types.CHANGE_LAYOUT,
        payload:nameLayout
    }
}


export const searchStory = (searchName) => {
    return {
        type: types.SEARCH_STORY,
        payload: searchName
    }
}

export const changeFlatForm = (orientation, issueTitle) => {
    return {
        type: types.BUTTON_CHANGE_FLATFORM,
        payload: {
            orientation: orientation,
            issueTitle: issueTitle
        }
    }
}

export const changeMode = (isMode) => {
    return {
        type: types.BUTTON_CHANGE_MODE,
        payload: isMode
    }
}

export const changeFontSize = (isSize) => {
    return {
        type: types.BUTTON_CHANGE_SIZE,
        payload: isSize
    }
}

export const changeScreenDevice = (isScreen, isValue) => {
    return {
        type: types.BUTTON_CHANGE_WIDTH,
        payload: {
            isScreen: isScreen,
            isValue: isValue

        }
    }
}


export const zoomFrame = (isZoom) => {
    return {
        type: types.ZOOM_FRAME,
        payload: isZoom
    }
}

export const changeLoadding = (isLoadding) => {
    return {
        type: types.CHANGE_LOADDING,
        payload: isLoadding
    }
}

export const changeSwipe = (currentPage) =>{
    return {
        type: types.CHANGE_SWIPE,
        payload: currentPage
    }
}

export const changeIFrame = (isURL, params, designPack, tocID, storyID, layout, orientation, mode, fontSize, navigate, isScreenFlatFrom, platform, issueTitle, layoutTiwg) => {
    return {
        type: types.CHANGE_URL_IFRAME,
        payload: {
            isURL: isURL,
            params: params,
            designPack: designPack,
            tocID: tocID,
            storyID: storyID,
            layout: layout,
            orientation: orientation,
            mode: mode,
            fontSize: fontSize,
            navigate: navigate,
            isScreenFlatFrom: isScreenFlatFrom,
            platform: platform,
            issueTitle: issueTitle,
            layoutTiwg:layoutTiwg
        }
    }
}
