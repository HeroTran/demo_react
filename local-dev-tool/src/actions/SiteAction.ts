import * as types from './types';
import * as api from '../api/cms';
import * as handleJson from '../utility/handleJson';

export function showAllIssue(params) {
    return (dispatch, getState) => {
        var promise = api.getIssues(params.id);
        dispatch(getListSiteRequest("waiting fecth for Site..."));
        return promise.then((result) => {
            dispatch(getListSiteSuccess(result,true));
        })
            .catch(e => {
                dispatch(getListSiteFailed("Failed fecth for Site...."))
            })
    }
}

export function getListSiteSuccess(issueList,isLoadding) {
    return {
        type: types.FETCH_DATA_SUCCCESS_SITE,
        payload:{
            issueList:issueList,
            isLoadding:isLoadding
        }
    };
}

export function getListSiteRequest(isRequest) {
    return {
        type: types.FETCH_DATA_REQUESTED_SITE,
        payload: isRequest
    };
}

export function getListSiteFailed(isFailed) {
    return {
        type: types.FETCH_DATA_FAILED_SITE,
        payload: isFailed
    };
}





/**
 * 
 * Show tab 2 option:download and not yet download
 */
export const  showTab = (isShow) => {
    return {
        type:types.SHOW_TAB,
        payload:isShow
    }
}




/**
 * 
 * event click button show with tab mobile,table,reload page
 */
export const  showFlatFrom = (isFlatForm) => {
    return {
        type:types.SHOW_FLAT_FORM,
        payload: isFlatForm
    }
}



export const  reloadPage = (isReload) => {
    return {
        type:types.SHOW_RELOAD_PAGE,
        payload: isReload
    }
}

/**
 * 
 * All event on magazine
 */

export function downLoadIssue(isDowload,siteId,issueId,idx) {
    return (dispatch, getState) => {
        var promise = api.downloadIssue(siteId,issueId);
        dispatch(downLoadRequest("waiting downloading..."));
        return promise.then((result) => {
            dispatch(downLoadSuccess(result,true,idx));
        })
        .catch(e => {
            dispatch(downLoadFailed("Failed download...."))
        })
    }
}

export const  downLoadSuccess = (result,isDowload,idx) => {
    return {
        type:types.DOWNLOAD_SUCCESS,
        payload:{
            result:result,
            isDowload:isDowload,
            idx:idx
        }
    }
}

export const  downLoadRequest = (isDowload) => {
    return {
        type:types.DOWNLOAD_REQUEST,
        payload:isDowload
    }
}

export const  downLoadFailed = (isDowload) => {
    return {
        type:types.DOWNLOAD_FAILED,
        payload:isDowload
    }
}




export const  eventView = (isView) => {
    return {
        type:types.EVENT_VIEW,
        payload: isView
    }
}


export function deleteIssue(isDowload,siteId,issueId,idx) {
    return (dispatch, getState) => {
        var promise = api.deleteIssue(siteId,issueId);
        dispatch(deleteRequest("waiting delete..."));
        return promise.then((result) => {
            dispatch(deleteSuccess(result,false,idx));
        })
            .catch(e => {
                dispatch(deleteFailed("Failed delete...."))
            })
    }
}

export const  deleteSuccess = (result,isDowload,idx) => {
    return {
        type:types.DELETED_ISSUE_SUCCESS,
        payload:{
            result:result,
            isDowload:isDowload,
            idx:idx
        }
    }
}

export const  deleteRequest = (isDelete) => {
    return {
        type:types.DELETED_ISSUE_REQUEST,
        payload:isDelete
    }
}

export const  deleteFailed = (isDelete) => {
    return {
        type:types.DELETED_ISSUE_FAILED,
        payload:isDelete
    }
}



export const  eventConfig = (isConfig,idx,idx_child) => {
    return {
        type:types.EVENT_CONFIG,
        payload:{
            isConfig:isConfig,
            idx:idx,
            idx_child:idx_child
        }
    }
}

export const  eventOpenModal = (idx) => {
    return {
        type:types.OPEN_MODAL,
        payload: idx
    }
}

export const  eventCloseModal = (idx) => {
    return {
        type:types.CLOSE_MODAL_CONFIG,
        payload: idx
    }
}


/**
 * Seach
 */
export const searchIssue = (name,siteId)=>{
    return {
        type:types.SEARCH_ISSUE,
        payload: {
            name:name,
            siteId:siteId
        }
    }
}

export const changeLoadding = (isLoadding)=>{
    return {
        type: types.CHANGE_LOADDING,
        payload:isLoadding
    }
}