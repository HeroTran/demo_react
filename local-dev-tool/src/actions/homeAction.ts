import * as types from './types';
import * as api from '../api/cms';
import * as handleJson from '../utility/handleJson';
import * as handleMethod from '../utility/handleMethod'; 

export function showListSite() {
    return (dispatch, getState) => {
        var promise = api.getSites();
        dispatch(getListSiteRequest("waiting fecth for Home..."));
        return promise.then((result) => {
            var data = handleJson.convertJsonToArray(result);
            handleMethod.setLocalStore("listSite",data);
            dispatch(getListSiteSuccess(data,true));
        })
            .catch(e => {
                dispatch(getListSiteFailed("failed..."))
            })
    }
}

export function getListSiteSuccess(site,isLoadding) {
    return {
        type: types.FETCH_DATA_SUCCCESS_MAGAZINE,
        payload:{
            site:site,
            isLoadding:isLoadding
        }
    };
}

export function getListSiteRequest(isRequest) {
    return {
        type: types.FETCH_DATA_REQUESTED_MAGAZINE,
        payload: isRequest
    };
}

export function getListSiteFailed(isFailed) {
    return {
        type: types.FETCH_DATA_FAILED_MAGAZINE,
        payload: isFailed
    };
}

export const showModal = (isShow) => {
    return {
        type: types.SHOW_MODAL,
        payload: isShow
    }
}

export const closeModal = (isHide) => {
    return {
        type: types.CLOSE_MODAL,
        payload: isHide
    }
}

export const submitConfig = (infoConfig) => {
    return {
        type: types.SUBMIT_CONFIG,
        payload: infoConfig
    }
}

export const onChangeInput = (propertyName, value) => {
    return {
        type: types.ON_CHANGE_INPUT,
        payload: {
            propertyName: propertyName,
            value: value
        }
    }
}

export const deleteSite = (siteID) => {
    return {
        type: types.DELETE_SITE,
        payload: siteID
    }
}

export const changeLoadding = (isLoadding)=>{
    return {
        type: types.CHANGE_LOADDING,
        payload:isLoadding
    }
}

export const searchSite = (name)=>{
    return {
        type:types.SEARCH_SITE,
        payload: name
    }
}