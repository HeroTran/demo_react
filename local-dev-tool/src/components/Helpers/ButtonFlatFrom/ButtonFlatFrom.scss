.Group-header{
	margin-bottom: 50px !important;
}
.right{
	float:right;
}
.Flatform{
	width: 100%;
	display: block;
	ul{
		float: right;
		li{
			display: inline-block;
			border: 1px solid #24abe1;
			margin: 0px 10px;
			padding: 5px;
			background: #fff;
			box-shadow: 1px 1px 1px #03A9F4;
			cursor: pointer;
			transition: all 0.5s ease;
			&:hover{
				transition: all 0.5s ease;
				background: #03a9f4;
    			box-shadow: 1px 1px 1px #fff;
				i{
					background-color: #03A9F4;
					color: #fff;
				}
			}
			i{
				font-size: 30px;
				color: #03A9F4;
			}
			&.active{
				background: #03a9f4;
    			box-shadow: 1px 1px 1px #fff;
				i{
					background-color: #03A9F4;
					color: #fff;
				}

			}
		}
	}
}