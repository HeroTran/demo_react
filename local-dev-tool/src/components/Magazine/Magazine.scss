.Main-Section{
	.Section{
		.Section-group{
			margin-top: 2em;
			.Group-header{
				span{
					position: absolute;
					z-index: 100;
					right: 0px;
					background-color: #00a6e2 !important;
					color: #fff;
					padding: 3px 10px;
					top: -15px;
					line-height: 30px;
					text-transform: uppercase;
					cursor: pointer;
					i{
						position: relative;
						top: 6px;
						left: -2px;
					}
				}
			}
			
			ul{
				max-width: 1150px;
				margin: 0 auto;
				padding: 0;
				display: block;
				-ms-flex-wrap: wrap;
				flex-wrap: wrap;
				list-style-type: none;
				li.Section-issue{
					display: inline-block;
					-ms-flex-direction: column;
					flex-direction: column;
					box-sizing: border-box;
					margin: 12px;
					/* margin-right: 1em; */
					box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
					transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
					border-bottom-left-radius: 2px;
					border-bottom-right-radius: 2px;
					width: 18%;
					&:hover{
						box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
						
					}
					a{
						text-decoration: none;
					}
					.Issue-cover{
						display: block;
						height: 242px;
						position: relative;
						overflow: hidden;
						color: #fff;
						text-align: center;
						background-color: #fff;
						border-top: 1px solid #bdbdbd;
						img{
							max-width: 100%;
							min-height: 180px !important;
							height: auto;
							vertical-align: top;
						}
						.Issue-cover-info{
							max-width: 100%;
							width: 100%;
							height: 270px;
							min-height: 200px;
							vertical-align: top;
							
							
						}
					}
					.Issue-detail{
						display: -ms-flexbox;
						display: flex;
						-ms-flex-direction: column;
						flex-direction: column;
						-ms-flex-align: center;
						align-items: center;
						box-sizing: border-box;
						height: 72px;
						border-top: 1px solid;
						border-color: rgba(0,0,0,.26);
						display: flex;
						-ms-flex-positive: 3;
						flex-grow: 3;
						-ms-flex-wrap: wrap;
						flex-wrap: wrap;
						-ms-flex-pack: center;
						justify-content: center;
						padding: 0.45em 0.75em;
						background-color: #ffffff;
						border-bottom-left-radius: 2px;
						border-bottom-right-radius: 2px;
						overflow: hidden;
						position: relative;
					}
				}
			}
		}
	}
	
}

.Issue-cover-overlay{
	text-align: center;
    background-color: transparent;
    position: absolute;
    display: -ms-flexbox !important;
    display: flex !important;
    -ms-flex-direction: column;
    flex-direction: column;
    -ms-flex-pack: center;
    justify-content: center;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}
.Section-issue:hover .Issue-cover-overlay {
    background-color: rgba(0,0,0,.75);
    transition: background-color 0.25s ease-in-out;
}

.Overlay-icons{
	opacity: 0;
	i{
		color: red !important;
	}
}
.Section-issue:hover .Overlay-icons {
    visibility: visible;
    transition: visibility 0.50s ease-in-out;
    transition-delay: 0.15s;
	cursor: pointer;
	opacity: 1;
}
.Issue-info li:first-child {
    color: #03A9F4;
    width: 100%;
    font-family: 'ProximaNova-Regular';
    display: -ms-flexbox;
    display: block;
    align-items: center;
    text-align: center;
    font-size: 16px;
    margin-bottom: 0px;
    overflow: hidden;
    text-overflow: ellipsis;

}
						
						
.Issue-info li {
    text-align: center;
    color: #FF5722;
    text-overflow: ellipsis;
	cursor: pointer;
}
.Issue-remove{
	.tit{
		height: 36px;
		background: #F44336;
		padding: 0px 0px;
		line-height: 30px;
		text-transform: uppercase;
		margin: 5px 0px;
		color: #fff;
		cursor: pointer;
		i{
			color: #fff !important;
			position: relative;
			top: 5px;
		}
	}
}

.search-site{
	float: right;
    width: 30%;
    position: relative;
    top: -10px;
	z-index: 100;
	input{
		border: 1px solid #03A9F4;
	}
}