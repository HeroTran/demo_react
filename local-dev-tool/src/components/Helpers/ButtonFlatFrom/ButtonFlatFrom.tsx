import * as React from 'react';
import * as cx from 'classnames';
import './ButtonFlatFrom.scss'

interface props{
    showFlatForm?:Function;
    searchIssue?:Function;
    params?:object;
    orientation?:string;
    
}
interface stateInfo{
}


export default class ButtonFlatFrom extends React.Component<props, stateInfo> {
 constructor(props:props){
        super(props);
  }
  
  handleShowFlatForm = (isFlatForm,event) => {
    var ul ;
    var li;
    if(event.target.classList.value.indexOf("material-icons") != -1){
      ul = event.target.parentElement.parentElement;
      li = event.target.parentElement;
    }else{
      ul  = event.target.parentElement;
      li = event.target;
    }
    ul.querySelector(".active").classList.remove("active");
    li.className += " active";
    this.props.showFlatForm(isFlatForm);
  }
  handleReloadPage = (event) =>{
     location.reload();
  }

  handleSearchIssue = (event,params) =>{
    console.log(event.target.value);
    const check = !/[~`!#$%\^&*+=\-\[\]\\';,/{}|\\":<>\?]/g.test(event.target.value.trim());
    if(check){
        let searchName = event.target.value.trim();
        this.props.searchIssue(searchName,params.id);
    }else{
        event.target.value = "";
    }
  }
  render() {
    const{showFlatForm,searchIssue,params} = this.props;
    
    return (
        <div className="wrap-button">
          <div className="left search-issue">
            <input type="text" onChange={(event)=>this.handleSearchIssue(event,this.props.params)}   className="button-searchInput form-control" placeholder="Search for Issue" />
          </div>
          <div className="right Flatform">
              <ul>
                <li className="active li-flatForm" onClick={this.handleShowFlatForm.bind(this,true)}><i className="material-icons md-12 dark">tablet_mac</i></li>
                <li className = "li-flatForm" onClick={this.handleShowFlatForm.bind(this,false)}><i className="material-icons md-12 dark">phone_iphone</i></li>
                <li onClick={this.handleReloadPage.bind(this)}><i className="material-icons md-12 dark">autorenew</i></li>
              </ul>
          </div>
        </div>
    );
  }
}

