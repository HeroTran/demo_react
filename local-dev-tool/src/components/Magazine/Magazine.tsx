import * as React from 'react'
import {Link} from 'react-router'
import './Magazine.scss'

type propsMagazine = {
    siteMagazine;
    deleteSite?:Function
}

export default class Magazine extends React.Component<propsMagazine,{}>{
    constructor(props:propsMagazine){
        super(props);
    }
    handleDeleteSite = (siteId,event) =>{
        //console.log(siteId);
        event.preventDefault();
        this.props.deleteSite(siteId);
    }
    render(){
        const{siteMagazine,deleteSite} = this.props;
        //console.log('siteMagazine',siteMagazine)
        return(
            <ul>
                {siteMagazine != null && siteMagazine.map((data, index) => {
                    var siteId = data.siteId;
                    return (
                        
                        <li className="Section-issue" key={index}>
                            <div className="Issue-cover">
                                {
                                    data.image ? 
                                    (<img src={data.image}/>)
                                    : (<img src="/images/logo.png"/>)
                                }
                                <div className="Issue-cover-overlay">
                                    <div className="Overlay-icons Issue-remove">
                                        <Link to={`/sites/${data.siteId}`}>
                                            <div className="tit">
                                                <i className="material-icons md-12 dark">visibility</i>
                                            </div>
                                        </Link>
                                        <a href={data.url} target="_blank">
                                            <div className="tit">
                                                <i className="material-icons md-12 dark">link</i> CMS
                                            </div>
                                        </a>
                                        
                                    </div>
                                </div>
                            </div>
                            <div className="Issue-detail">
                                <ul className="Issue-info">
                                    <li className="info-name">{data.name}</li>
                                    <li className="info-id">{data.siteId}</li>
                                </ul>
                                
                            </div>
                    </li> 
                );
        })}

            </ul>
        );
    }
    
};