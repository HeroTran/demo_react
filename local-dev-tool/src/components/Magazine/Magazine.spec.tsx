import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Magazine from './Magazine';
import * as siteAction from '../../actions/SiteAction';
import ButtonFlatFrom from '../../components/Helpers/ButtonFlatFrom/ButtonFlatFrom';


const testMagazine = [
    {
        "siteId":"987",
        "name": "PC World",
        "designPack": "dp-PCWorld",
        "url": "http://pcworld-us.audiencemedia.com/api/amm/v1/auth-issues",
        "image": "images/magazines/pcworld.jpg"
    }
]

describe('Magazine component', () => {
    it('render with correct UI', () => {
        const component = renderer.create(<Magazine siteMagazine={testMagazine}/>).toJSON();
        expect(component).toMatchSnapshot();
    })
    it('render correctly number of magazines', () => {
        const component =mount(<Magazine siteMagazine={testMagazine}/>);
        expect(component.find('.Section-issue')).toHaveLength(1);
    })
    it('render with correct thumbnail of magazine', () => {
        const component =mount(<Magazine siteMagazine={testMagazine}/>);
        expect(component.find('.Issue-cover img').prop('src') !== '/images/logo.png').toBeTruthy();
    })
    it('render correctly with default thumbnail', () => {
        const noThumb = [...testMagazine];
        noThumb[0].image = null;
        const component =mount(<Magazine siteMagazine={noThumb}/>);
        expect(component.find('.Issue-cover img').prop('src') === '/images/logo.png').toBeTruthy();
    })
    it('each Magazine has two button View and CMS', () => {
        const component = shallow(<Magazine siteMagazine={testMagazine}/>);
        expect(component.find('.Issue-remove a')).toHaveLength(1);
    })
    it('info of Magazine should renders name and siteID', () => {
        const component = mount(<Magazine siteMagazine={testMagazine}/>);
        expect(component.find('.Issue-info li')).toHaveLength(2);
    })
});