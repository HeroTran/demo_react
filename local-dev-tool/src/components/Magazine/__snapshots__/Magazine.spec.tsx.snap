// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Magazine component render with correct UI 1`] = `
<ul>
  <li
    className="Section-issue"
  >
    <div
      className="Issue-cover"
    >
      <img
        src="images/magazines/pcworld.jpg"
      />
      <div
        className="Issue-cover-overlay"
      >
        <div
          className="Overlay-icons Issue-remove"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="tit"
            >
              <i
                className="material-icons md-12 dark"
              >
                visibility
              </i>
            </div>
          </a>
          <a
            href="http://pcworld-us.audiencemedia.com/api/amm/v1/auth-issues"
            target="_blank"
          >
            <div
              className="tit"
            >
              <i
                className="material-icons md-12 dark"
              >
                link
              </i>
               CMS
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="Issue-detail"
    >
      <ul
        className="Issue-info"
      >
        <li
          className="info-name"
        >
          PC World
        </li>
        <li
          className="info-id"
        >
          987
        </li>
      </ul>
    </div>
  </li>
</ul>
`;
